Factory/DiactorosFactory.php

```diff
@@ -0,0 +1,135 @@
+ */
+class DiactorosFactory implements HttpMessageFactoryInterface
+{
+    public function __construct()
+    {
+        if (!class_exists('Zend\Diactoros\ServerRequestFactory')) {
+            throw new \RuntimeException('Zend Diactoros must be installed to use the DiactorosFactory.');
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function createRequest(Request $symfonyRequest)
+    {
+        $request = DiactorosRequestFactory::fromGlobals(
+            $symfonyRequest->server->all(),
+            $symfonyRequest->query->all(),
+            $symfonyRequest->request->all(),
+            $symfonyRequest->cookies->all(),
+            $this->getFiles($symfonyRequest->files->all())
+        );
+
+        try {
+            $stream = new DiactorosStream($symfonyRequest->getContent(true));
+        } catch (\LogicException $e) {
+            $stream = new DiactorosStream('php://temp', 'wb+');
+            $stream->write($symfonyRequest->getContent());
+        }
+
+        $request = $request->withBody($stream);
+
+        foreach ($symfonyRequest->attributes->all() as $key => $value) {
+            $request = $request->withAttribute($key, $value);
+        }
+
+        return $request;
+    }
+
+    /**
+     * Converts Symfony uploaded files array to the PSR one.
+     *
+     * @param array $uploadedFiles
+     *
+     * @return array
+     */
+    private function getFiles(array $uploadedFiles)
+    {
+        $files = array();
+
+        foreach ($uploadedFiles as $key => $value) {
+            if ($value instanceof UploadedFile) {
+                $files[$key] = $this->createUploadedFile($value);
+            } else {
+                $files[$key] = $this->getFiles($value);
+            }
+        }
+
+        return $files;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function createUploadedFile(UploadedFile $symfonyUploadedFile)
+    {
+        return new DiactorosUploadedFile(
+            $symfonyUploadedFile->getRealPath(),
+            $symfonyUploadedFile->getSize(),
+            $symfonyUploadedFile->getError(),
+            $symfonyUploadedFile->getClientOriginalName(),
+            $symfonyUploadedFile->getClientMimeType()
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function createResponse(Response $symfonyResponse)
+    {
+        $stream = new DiactorosStream('php://temp', 'wb+');
+        if ($symfonyResponse instanceof StreamedResponse) {
+            ob_start(function ($buffer) use ($stream) {
+                $stream->write($buffer);
+
+                return false;
+            });
+
+            $symfonyResponse->sendContent();
+            ob_end_clean();
+        } else {
+            $stream->write($symfonyResponse->getContent());
+        }
+
+        $response = new DiactorosResponse(
+            $stream,
+            $symfonyResponse->getStatusCode(),
+            $symfonyResponse->headers->all()
+        );
```

Comment:
The conversion is incomplete here:

- you are missing the status text (although it might be a pain as we don't have a getter for it)
- you are missing cookies